example.php

```diff
@@ -0,0 +1,153 @@
+# A comment your comment
+from graphics import *
+from time import sleep
+from math import sin, cos
+import argparse
+import random
+import sys
+
+# Defaults
+ratio = 3
+r = 300
+screen_size = r * 2
+NUM_CIRCLES = 7
+colours = []
+ratios = []
+
+
+
+def generate(circle, colour):
```

Comment:
```diff
- def generate(circle, colour):
+ # A comment {{ .funcComment }}
```

Change `{{ .funcComment }}` by commenting: `@CodeLingoBot rewrite set funcComment "your comment"`

Reply:
@CodelingoBotTester rewrite set funcComment "your comment"

Reply:
@CodelingoBotTester rewrite set funcComment "your comment"

Reply:
strconv.Atoi: parsing "object\ndiff --git a/circle.py b/circle.py\nindex bd4e74f..f773520 100644\n--- a/circle.py\n+++ b/circle.py\n@@ -1,4 +1,4 @@\n-import types\n+# A comment your comment\r\n from graphics import *\n from time import sleep\n from math import sin, cos\n@@ -16,7 +16,7 @@\n \n \n \n-def generate(circle, colour):\n+# A comment your comment\r\n     circles = []\n     r_small = circle.radius / ratio\n     r2 = r_small * 2\n@@ -149,3 +149,5 @@ def has_stdin():\n print(f\"Completed in {run_time} seconds\")\n win.getMouse()\n win.close()\n+\n+# A comment at the end\n\\ No newline at end of file\ndiff --git a/example.php b/example.php\nnew file mode 100644\nindex 0000000..fbd9d85\n--- /dev/null\n+++ b/example.php\n@@ -0,0 +1,153 @@\n+# A comment your comment\r\n+from graphics import *\n+from time import sleep\n+from math import sin, cos\n+import argparse\n+import random\n+import sys\n+\n+# Defaults\n+ratio = 3\n+r = 300\n+screen_size = r * 2\n+NUM_CIRCLES = 7\n+colours = []\n+ratios = []\n+\n+\n+\n+def generate(circle, colour):\n+    circles = []\n+    r_small = circle.radius / ratio\n+    r2 = r_small * 2\n+    # 60 degrees in radians for math functions\n+    angles = [1.0472, -1.0472]\n+    center = circle.getCenter()\n+    Cx = center.getX()\n+    Cy = center.getY()\n+\n+    # Generate 3 horizontal circles\n+    for i in range(0, 3):\n+        temp = Circle(Point((Cx + ((i-1) * r2)), Cy), r_small)\n+        temp.setFill(colour)\n+        circles.append(temp)\n+\n+    # Generate the 4 \"corner\" circles\n+    for i in range(0, 2):\n+        for angle in angles:\n+            if i == 0:\n+                X = Cx - (r2 * cos(angle))\n+                Y = Cy - (r2 * sin(angle))\n+            else:\n+                X = Cx + (r2 * cos(angle))\n+                Y = Cy + (r2 * sin(angle))\n+            center = Point(X, Y)\n+            temp = Circle(center, r_small)\n+            temp.setFill(colour)\n+            circles.append(temp)\n+    return circles\n+\n+\n+# Print all circles in resultant list of circles\n+def draw_circles(circles):\n+\n+    count = 0\n+    for c in circles:\n+        c.draw(win)\n+        #sleep(.3)\n+        count = count + 1\n+    print(f\"{count} circles drawn\")\n+\n+\n+def generate_colours():\n+    colours = []\n+\n+    for i in range(0, layers):\n+        c = color_rgb(random.randint(0, 255),\n+                      random.randint(0, 255),\n+                      random.randint(0, 255))\n+        colours.append(c)\n+    return colours\n+\n+def run(colours):\n+    global ratios, ratio\n+    if not colours:\n+        colours = generate_colours()\n+    if ratios:\n+        ratio = ratios[0]\n+        ratios = ratios[1:]\n+        \n+   \n+    # Create initial circle in list\n+    initial_circle = Circle(Point(r, r), r)\n+    initial_circle.setFill(colours[0])\n+    circles = [initial_circle]\n+    curr_circles = circles\n+   \n+    for i in range(0, layers):\n+        print(f\"Generation {i}:\")\n+        draw_circles(curr_circles)\n+        if ratios:\n+            ratio = ratios[0]\n+            ratios = ratios[1:]\n+            \n+        if i == layers - 1:\n+            break\n+        curr_circles = []\n+\n+        # sleep(1)\n+        for j in range(len(circles) - (NUM_CIRCLES**i), len(circles)):\n+            curr_circles.extend(generate(circles[j], colours[i+1]))\n+        circles.extend(curr_circles)\n+\n+\n+\n+\n+def has_stdin():\n+    return not sys.stdin.isatty()\n+\n+\n+# --------- Begin ----------\n+\n+\n+print(\"Welcome to Circles\")\n+\n+# Set up command line args\n+parser = argparse.ArgumentParser()\n+parser.add_argument(\"-g\", \"--generation\", default=\"3\", help=\"set number of generations\", type=int)\n+args = parser.parse_args()\n+layers = args.generation\n+\n+\n+if has_stdin():\n+    count = 0\n+    print(\"Running with input file. -g flag will be ignored, if specified\")\n+    for line in sys.stdin:\n+        line = line.strip().split(\" \")\n+        if \"/\" in line[0]:\n+            line[0] = int(line[0][-1])\n+        else:\n+            line[0] = int(1/int(float(line[0])))\n+        ratios.append(line[0])\n+        colours.append(color_rgb(int(line[1]), int(line[2]), int(line[3])))\n+        count = count + 1\n+    layers = count\n+else:\n+    print(\"Running with defaults\")\n+\n+print(f\"Running {layers} generations\")\n+print(ratios)\n+\n+# Display window\n+win = GraphWin(\"Circles\", screen_size, screen_size)\n+\n+run_time = time.time()\n+\n+run(colours)\n+\n+run_time = time.time() - run_time\n+print(f\"Completed in {run_time} seconds\")\n+win.getMouse()\n+win.close()\n+\n+# A comment at the end\n\\ No newline at end of file\n\n152": invalid syntax

---
<details>
<summary>CodeLingoBot commands and options</summary>

You can trigger CodeLingo actions by commenting on this PR:

+ `@CodelingoBotTester  review` reviews the PR based on default Tenets for the repo's language or custom Tenets added in a codelingo.yaml file in the root of the repo. <details><summary>You can trigger CodeLingo actions by replying to comments generated by`@CodelingoBotTester  review`</summary>
    + `@CodelingoBotTester  review ignore [reason]` ignores the issue for current and future reviews with an optional reason why.
    + `@CodelingoBotTester  review un-ignore` un-ignores the issue for current and future reviews.
+ `@CodelingoBotTester  rewrite` makes a PR to this PR fixing the issues found based on default Tenets for the repo's language or custom Tenets added in a codelingo.yaml file in the root of the repo. <details><summary>You can trigger CodeLingo actions by replying to comments generated by`@CodelingoBotTester  rewrite`</summary>

    + `@CodelingoBotTester  rewrite set <varname>` sets the value of a variable in a rewrite template.

Mention @CodelingoBotTester  to reach us.

</details>